### DISCUSSION TITLE
Example of React to Solid port

### DISCUSSION THREAD

**Discussion by edemaine** [solidjs-community/solid-codemod#4] [General]
I just ported a ~6,000-line CoffeeScript app (shared whiteboard) from React to Solid. I'm hoping it serves as a useful example of what could be mostly automated by a codemod.

Here's the commit, which is mostly the port though a few other changes as well: https://github.com/edemaine/cocreate/commit/554cd8619fb10af23d0565bbd0514fc92759fb49
This includes everything I needed to change, including eslint and Meteor settings.

Some patterns I noticed, many of which would be easy to automate:
* `useEffect` with return function converted into `createEffect` or `onMount` with `onCleanup` (#1)
* Not sure exactly what `useRenderEffect` should correspond to; in some cases I used `createRenderEffect` to force early evaluation, but it's really closer to `createEffect` so I often used that.
* Removing `useRef` and `.current` (#3)
* `useState` to `createSignal` and adding lots of parentheses (#1)
* Lots of renaming converting from `react-router-dom` to `solid-app-router`. For example, `path` is called `pathname`, `useHistory` is roughly equivalent to `useNavigate`, etc.
* `useTracker` to `createTracker` and `react-meteor-data` to `solid-meteor-data` (this is Meteor specific)
* `className` to `class` (for cleaner code)
* Undestructuring of props, `splitProps`, etc. (#5)
* Drop `React.memo` everywhere
* Gotcha: `style={undefined}` behaves very differently between React (which just deletes defined styles) and Solid (which deletes everything).

**Comment by felixfeixu** [comment]
Just out of my curiosity, any performance benefit gained from the porting?
